spec/e2e/association_proxy_spec.rb

```diff
@@ -49,6 +83,39 @@
     billy.favorite_lesson = math
   end
 
+  it 'Should allow for string parameter with varibale length relationship notation' do
+    post
+    expect_queries(1) do
+      post.comments.with_associations('owner.knows*').each do |comment|
+        comment.owner.knows.each do |known|
+          known.knows.to_a
+        end
+      end
+    end
+  end
+
+  it 'Should allow for string parameter with fixed length relationship notation' do
+    post
+    expect_queries(1) do
+      post.comments.with_associations('owner.knows*2').each do |comment|
+        comment.owner.knows.each do |known|
+          known.knows.to_a
+        end
+      end
+    end
+  end
+
+  it 'Should allow for string parameter with varibale length relationship notation' do
+    user = post.owner.as(:owner)
```

Comment:
```suggestion
    user = post.owner(chainable: true)
```
avoids a query. The `owner` call in this form could go inside `expect_queries`.